### ISSUE TITLE
Feature Request: Use websockets for synchronizing changes from server automatically

### ISSUE THREAD

**Issue by riclucio** [nhat-phan/merge-request-integration#36]
It looks like every time a change is done on the merge request (a new comment added, approval/rejection, new commit, etc.), we always have to click on a refresh button.
Possibly it could leverage WebSocket for pulling changes to the merge request automatically? (not sure if GitLab supports WebSocket connections though)

If this is done, it would also be nice to show notifications in the IDE for when comments are added/approvals/rejections happen in the merge request, so we don't need to refresh each merge request from time to time.

**Comment by nhat-phan** [comment]
Thank you for your feedback.

Gitlab doesn't have web socket api, but refresh every change is possible.

**Comment by riclucio** [comment]
In this case how would you update changes from other people in the merge request?
Pooling at some interval?

**Comment by nhat-phan** [comment]
Yeah, pooling at some interval in a separated thread is the only way to keep your data updated. Then when you select the merge request, the plugin will run `git fetch` to get changes. The downside is it might not work if the branch was rebased.

There is another approach which does not depend on git locally. That's get all changes from gitlab (from head and target branch) then compare them and display diff. The problems of this approach is it's complicated and you can't navigate through the source code (the same problem in #33 ).

**Comment by riclucio** [comment]
By "changes" I mean new comments / approvals.
If those events could refresh automatically in the IDE it would already help a lot.
New checkins could be indicated by some red icon (along with notification), then this can be pulled again when I click "Code Review".
What do you think?
